crates/sui-core/src/quorum_driver/mod.rs

```diff
@@ -714,29 +718,48 @@ where
 }
 
 // TODO: categorize all possible SuiErrors
-fn convert_to_quorum_driver_error_if_nonretryable(
-    err: SuiError,
-    tx_digest: &TransactionDigest,
-    action: &'static str,
-) -> Option<QuorumDriverError> {
-    match &err {
+fn convert_to_quorum_driver_error_if_nonretryable(err: &SuiError) -> Option<QuorumDriverError> {
+    match err {
         // TODO: rewrite the equivocation detection code to make it more deterministic
         SuiError::QuorumFailedToProcessTransactionWithConflictingTransactions {
             conflicting_txes,
             retried_tx_digest,
             retried_tx_success,
+        } => Some(QuorumDriverError::ObjectsDoubleUsed {
+            conflicting_txes: conflicting_txes.clone(),
+            retried_tx: *retried_tx_digest,
+            retried_tx_success: *retried_tx_success,
+        }),
+        SuiError::QuorumFailedToProcessTransaction {
+            good_stake: _,
+            errors,
+            conflicting_tx_digests,
         } => {
-            debug!(?tx_digest, "Got unretryable error when {action}: {err}");
-            Some(QuorumDriverError::ObjectsDoubleUsed {
-                conflicting_txes: conflicting_txes.clone(),
-                retried_tx: *retried_tx_digest,
-                retried_tx_success: *retried_tx_success,
-            })
-        }
-        _ => {
-            debug!(?tx_digest, "Got retryable error when {action}: {err}");
-            None
+            if !conflicting_tx_digests.is_empty() {
+                return None;
+            }
+
+            // All the errors are the same, pick a sentinel error
+            let error = match &errors[..] {
+                [error, errors @ ..] if errors.iter().all(|e| e == error) => error,
+                _ => return None,
+            };
```

Comment:
the error vector is not sorted so this won't work reliably, e.g. it could be `[DependentPackageNotFound, RpcError]`.

Reply:
In this case, I don't think order is relevant, because this code is checking that all the errors in `errors` are the same, but this would also fail with the input that you described.  My thinking was that if there was any doubt about the error being retryable, we should retry, and only when all the validators agreed that it was the same error would it bail -- but the hotfix you mentioned above makes a lot of sense to me, so I'll go for that!